src/btreemap.rs

```diff
@@ -400,9 +396,13 @@ impl<M: Memory + Clone, K: Storable, V: Storable> BTreeMap<M, K, V> {
         self.length
     }
 
-    /// Returns a reference to the memory used by the map.
-    pub fn get_memory(&self) -> M {
-        self.memory.clone()
+    /// Returns the underlying memory.
+    pub fn forget(self) -> M {
+        self.allocator.forget()
+    }
```

Comment:
I think the name `forget` here can be misleading, since no data is actually forgotten. Consider this code:

```
let btree: BTreeMap<u64, u64. _> = BTree::init(mem);
btree.insert(1,1).unwrap();
let mem = btree.forget();
let btree: BTreeMap<u64, u64. _> = BTree::init(mem);
assert_eq!(btree.len(), 0); // panics
```

Given the name `forget`, I'd assume that this deletes the btree from memory, and it leads to the, IMO, the unintuitive result of a newly initialized btree not being being empty.

Another example:

```
let btree = BTree::init_with_sizes(10, 10, mem);
let mem = btree.forget();
let btree = BTree::init_with_sizes(20, 20, mem); // will panic
```

I'd also, judging by the name `forget`, assume that my initial btree doesn't exist anymore and I can initialize a new one in its place, but that's not what will happen. The underlying btree is still intact, and trying to re-initialize a tree with larger keys/values panics.

Reply:
Here, `forget` means "forget that you have a map in get me back the raw memory". If it was supposed to delete the contents, I'd call it "clear()" or "erase()`.
We use this name in other data structures as well. Do you think `into_memory()` would be a better name?

Reply:
I think `into_memory` is much clearer, thanks!